rest/specs/src/main/filtered-resources/openapi/openapi.yaml

```diff
@@ -194,32 +194,18 @@ components:
       description: Transformation request.
       type: object
       required:
-        - id
-        - stagedLocation
-        - productLocation
+        - location
         - mimeType
         - bytes
-        - callbackUrl
       properties:
-        id:
-          type: string
-          description: The unique identifier for a request.
-          minLength: 1
-          maxLength: 255
-          pattern: '^[a-zA-Z0-9]*$'
-          example: 30f14c6c1fc85cba12bfd093aa8f90e3
-        stagedLocation:
-          type: string
-          description: The temporary location of the ingested product.
-          minLength: 1
-          maxLength: 2048
-          example: https://bucket.s3.amazonaws.com
-        productLocation:
+        location:
           type: string
-          description: The final location of the ingested product.
+          description: >
+            The location of the ingested product. This URL must support HTTP GET at location/{productId} to retrieve
+            the product and HTTP PUT with attachment at location/{productId}/{metadataType} to store metadata.
           minLength: 1
```

Comment:
✏️ I think you can specify `format: uri`

Reply:
Nice! I'll add that.

Reply:
A word of caution here. `uri` is _not_ an officially supported `format` for `string` and may not be handled properly by the API validator or code generator. See https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#dataTypes.

Reply:
That'll be fixed once we create our own generators? For now, I would still leave it there as it better documents the restrictions.

Reply:
Looks like I have to remove `format: uri` -> @lessarderic's prophesy was correct